improvements to github exporter docs (#166)
port-labs/port-docs: docs/exporters/github-exporter/exporting-all-repositories.md

---

> sidebar_position: 3

Reviewer: Again, verify sidebar position numbering

title: Exporting all repositories
---

:::note Prerequisites

- [Please install our GitHub app](./installation).

:::

# Exporting all repositories

> By the end of this tutorial, we will have a `microservice` Blueprint that contains the properties `repository URL` and `Description` which are automatically synced from GitHub to Port entity.

Reviewer 2: ```suggestion
IN this tutorial you will learn how to create a `microservice` Blueprint that contains the properties `repository URL` and `Description`, which are automatically synced from GitHub to Port Entity.
```

1. Create a `microservice` Blueprint and `port-app-config.yml` configuration file.

To export your GitHub `repositories` to Port, you can use the following Port Blueprints definitions, and `port-app-config.yml`:

<details>
<summary> Microservice Blueprint </summary>

```json showLineNumbers
{
  "identifier": "microservice",
  "title": "Microservice",
  "icon": "Microservice",
  "schema": {
    "properties": {
      "description": {
        "title": "Description",
        "type": "string",
        "format": "markdown"
      },
      "url": {
        "title": "URL",
        "format": "url",
        "type": "string"
      }
    },
    "required": []
  },
  "mirrorProperties": {},
  "calculationProperties": {},
  "relations": {}
}
```

</details>

> In order to apply the `port-app-config.yml` in the whole organization, place it in the `.github` folder or in the `.github-private` repository in the root directory.

Reviewer 2: ```suggestion
To apply the `port-app-config.yml` in the whole organization, place it in the `.github` folder or in the `.github-private` repository in the root directory.
```
